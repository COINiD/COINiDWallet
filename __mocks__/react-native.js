import MockAsyncStorage from 'mock-async-storage';

import { WebSocket } from 'mock-socket';

const rn = require('react-native');

global.fetch = require('jest-fetch-mock');

global.fetch.mockResponse(JSON.stringify({ data: '12345' }));

global.WebSocket = WebSocket;

jest.useFakeTimers();

jest.mock('Alert', () => ({ alert: jest.fn() }));

jest.mock('AsyncStorage', () => new MockAsyncStorage());

jest.mock('NativeAnimatedHelper');

jest.mock('NativeModules', () => ({
  P2PTransferBLECentralModule: {
    addListener: jest.fn(),
  },
  P2PTransferBLEPeripheralModule: {},
  KeyboardObserver: {
    addListener: jest.fn(),
  },
  StatusBarManager: {
    HEIGHT: 42,
    setColor: jest.fn(),
    setStyle: jest.fn(),
    setHidden: jest.fn(),
    setNetworkActivityIndicatorVisible: jest.fn(),
    setBackgroundColor: jest.fn(),
    setTranslucent: jest.fn(),
  },
}));

module.exports = rn;
