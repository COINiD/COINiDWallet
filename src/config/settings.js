export default {
  coin : 'testnet',
  appReturnScheme : 'coinid-tbtc',
  coldWalletMode: true,
  language: 'system',
  availableLanguages : ['system','en','de','fr','es','pt','eo','sv'],
  currency: 'USD',
  availableCurrencies : ['USD','EUR','JPY','GBP','CAD','CHF','KRW','NZD','SEK','NOK','DKK'],
  ranges: ['day', 'week', 'month', 'year'],
  refreshRates: {
    exchangeRate: 600000, // 10 minutes
    account: 600000, // 10 minutes
  },
  coldTransportTypes: [
    {
      key: '',
      title: 'settings.coldtransport.showdialog.title',
      description: 'settings.coldtransport.showdialog.description',
    },
    {
      key: 'qr',
      title: 'settings.coldtransport.qrcodes.title',
      description: 'settings.coldtransport.qrcodes.description',
    },
    {
      key: 'ble',
      title: 'settings.coldtransport.ble.title',
      description: 'settings.coldtransport.ble.description',
    },
  ],
  lockDurations: [
    { milliseconds: 0, title: 'settings.lockduration.immediately' },
    { milliseconds: 60000, title: 'settings.lockduration.1minute' },
    { milliseconds: 300000, title: 'settings.lockduration.5minutes' },
    { milliseconds: 900000, title: 'settings.lockduration.15minutes' },
    { milliseconds: 1800000, title: 'settings.lockduration.30minutes' },
    { milliseconds: 3600000, title: 'settings.lockduration.1hour' },
    { milliseconds: 14400000, title: 'settings.lockduration.4hours' },
  ],
  aboutUrl: 'https://coinid.org',
  feedbackUrl: 'https://coinid.org/feedback',
  guidesUrl: 'https://coinid.org/guides',
  telegramUrl: 'https://t.me/joinchat/IARCoBAdhQOIEN_7u941Qg',
  twitterUrl: 'https://twitter.com/COINiDGroup',
  instagramUrl: 'https://www.instagram.com/coinidgroup',
  offlineGuideUrl: {
    ios: 'https://coinid.org/guides/preparing-ios-cold-wallet',
    android: 'https://coinid.org/guides/preparing-android-cold-wallet',
  },
  isTestnet : true,
};
