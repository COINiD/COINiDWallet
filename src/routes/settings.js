import { createStackNavigator } from 'react-navigation';
import SettingsRoute from '../components/SettingsRoute';

export const settingRoutes = {
  Home: {
    screen: SettingsRoute,
    swipeEnabled: false,
    title: 'Settings',
    isHome: true,
  },
  Passcode: {
    screen: SettingsRoute,
    title: 'Require unlocking',
  },
  OfflineTransport: {
    screen: SettingsRoute,
    title: 'Offline transport',
  },
  PreferredCurrency: {
    screen: SettingsRoute,
    title: 'Preferred currency',
  },
  SignMessage: {
    screen: SettingsRoute,
    title: 'Sign message',
  },
  About: {
    screen: SettingsRoute,
    title: 'About',
  },
  AccountList: {
    screen: SettingsRoute,
    title: 'Select an account',
  },
  AccountInformation: {
    screen: SettingsRoute,
    title: 'Account information',
  },
};

export const getSettingsNavigator = cardStyle => createStackNavigator(settingRoutes, {
  animationEnabled: true,
  headerMode: 'none',
  cardStyle,
});
