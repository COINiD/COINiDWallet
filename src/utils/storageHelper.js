/**
 * Helper for storage
 */

import { AsyncStorage } from 'react-native';

const getStorageItem = (namespace, key) => AsyncStorage.getItem(`@${namespace}:${key}`).then(JSON.parse);

const setStorageItem = (namespace, key, value) => {
  const val = JSON.stringify(value);
  return AsyncStorage.setItem(`@${namespace}:${key}`, val);
};

const removeStorageItem = (namespace, key) => AsyncStorage.removeItem(`@${namespace}:${key}`);

const reset = namespace => AsyncStorage.getAllKeys().then((keys) => {
  const filteredKeys = keys.filter(key => key.startsWith(`@${namespace}`));
  return AsyncStorage.multiRemove(filteredKeys);
});

export default namespace => ({
  get: key => getStorageItem(namespace, key),
  set: (key, value) => setStorageItem(namespace, key, value),
  remove: key => removeStorageItem(namespace, key),
  reset: () => reset(namespace),
});
