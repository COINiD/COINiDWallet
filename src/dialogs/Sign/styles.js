import { StyleSheet } from 'react-native';
import { colors, fontWeight } from '../../config/styling';
import styleMerge from '../../utils/styleMerge';
import parentStyles from '../dialogStyles';

export default styleMerge(
  parentStyles('light'),
  StyleSheet.create({
    summaryContainer: {
      width: '100%',
      backgroundColor: 'rgba(255, 255, 255, 0.95)',
      zIndex: 100,
      position: 'relative',
      marginBottom: -16,
      paddingBottom: 16,
    },
    batchedHeaderContainer: {
      marginTop: -16,
      paddingTop: 16,
      backgroundColor: 'rgba(255, 255, 255, 0.95)',
      width: '100%',
      zIndex: 100,
      position: 'relative',
    },
    batchedHeader: {
      color: colors.getTheme('light').fadedText,
      marginBottom: 8,
      ...fontWeight.medium,
    },
  })
);
