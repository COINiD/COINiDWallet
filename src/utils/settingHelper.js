'use strict';

/**
 * Helper for loading and saving settings.
 */

import { EventEmitter } from 'events';
import storageHelper from './storageHelper';
import Settings from '../config/settings';

class SettingHelper extends EventEmitter {
  constructor(coin) {
    super();

    this.setMaxListeners(50);

    this.storageNS = `${coin}:settings`;
    this.storage = storageHelper(this.storageNS);
    this.defaultSettings = {
      coldWalletMode: true,
      currencies: Settings.availableCurrencies,
      currency: Settings.currency,
      range: 0,
      usePasscode: true,
      lockAfterDuration: 60000,
      preferredColdTransport: '',
    };
    this.settings = { ...this.defaultSettings };
  }

  load = () => {
    const getDefaultIfEmpty = (key, data) =>
      data !== null ? data : this.defaultSettings[key];

    var p = Object.entries(this.defaultSettings).map(([key, value]) =>
      this.storage.get(key).then(data => getDefaultIfEmpty(key, data))
    );

    return Promise.all(p).then(
      ([
        coldWalletMode,
        currencies,
        currency,
        range,
        usePasscode,
        lockAfterDuration,
        preferredColdTransport,
      ]) => {
        this.settings = {
          coldWalletMode,
          currencies,
          currency,
          range,
          usePasscode,
          lockAfterDuration,
          preferredColdTransport,
        };

        this.emit('updated', this.settings);
      }
    );
  };

  update = (key, value) => {
    this.settings[key] = value;
    this.save();
    this.emit('updated', this.settings);
  };

  save = () => {
    var p = Object.entries(this.settings).map(([key, value]) => {
      return this.storage.set(key, value);
    });

    return Promise.all(p);
  };

  reset = () => {
    this.storage.reset();
  };

  getAll = () => {
    return this.settings;
  };
}

let settingHelpersCache = {}; // for local caching so we dont create several for same coin.

module.exports = function(coin) {
  if (settingHelpersCache[coin] === undefined) {
    settingHelpersCache[coin] = new SettingHelper(coin);
  }

  return settingHelpersCache[coin];
};
