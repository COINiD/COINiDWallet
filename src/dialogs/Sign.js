import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Alert, View } from 'react-native';
import Big from 'big.js';

import {
  Text,
  BatchList,
  RowInfo,
  FeeSlider,
  COINiDTransport,
  Button,
  CancelButton,
} from '../components';

import { numFormat } from '../utils/numFormat';

import { colors, fontWeight, fontSize } from '../config/styling';
import styleMerge from '../utils/styleMerge';
import parentStyles from './styles/common';

import WalletContext from '../contexts/WalletContext';

const styles = styleMerge(
  parentStyles('light'),
  StyleSheet.create({
    summaryContainer: {
      width: '100%',
      zIndex: 100,
      position: 'relative',
      marginBottom: -16,
      paddingBottom: 16,
      backgroundColor: colors.getTheme('light').seeThrough,
    },
    batchedHeaderContainer: {
      width: '100%',
      zIndex: 100,
      position: 'relative',
      backgroundColor: colors.getTheme('light').seeThrough,
    },
    batchedHeader: {
      marginBottom: 8,
      ...fontWeight.medium,
    },
    totalError: { color: '#FA503C' },
    warningText: {
      fontSize: fontSize.small,
      color: colors.getTheme('light').warning,
      marginTop: -16,
      marginBottom: 16,
    },
  }),
);

class Sign extends Component {
  static contextType = WalletContext;

  static propTypes = {
    payments: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
    balance: PropTypes.number.isRequired,
    realBalance: PropTypes.number.isRequired,
    navigation: PropTypes.shape({}).isRequired,
    dialogRef: PropTypes.shape({}).isRequired,
    onAddToBatch: PropTypes.func.isRequired,
    onRemoveFromBatch: PropTypes.func.isRequired,
    onQueuedTx: PropTypes.func.isRequired,
  };

  constructor(props, context) {
    super(props);

    this.fee = 0;
    const { ticker } = context.coinid;

    this.state = {
      total: 0,
      subTotal: 0,
      fee: this.fee,
      ticker,
    };
  }

  componentDidMount() {
    this._calculateTotal();
  }

  componentWillReceiveProps() {
    this._calculateTotal();
  }

  /*
  shouldComponentUpdate() {
    if (this.forceNoRender) {
      return false;
    }
    return true;
  }
  */

  _calculateTotal = () => {
    const { payments } = this.props;
    const { fee } = this;

    let subTotal = payments.reduce((a, c) => a.plus(c.amount), Big(0));
    const total = Number(subTotal.plus(fee));
    subTotal = Number(subTotal);

    this.setState({
      total,
      subTotal,
      fee,
    });
  };

  _verify = () => {
    let { total } = this.state;
    total = Number(total);

    const { payments, realBalance } = this.props;
    const errors = [];

    if (!payments.length) {
      errors.push({
        type: 'payments',
        message: 'No payments...',
      });
    }

    if (isNaN(total)) {
      errors.push({
        type: 'amount',
        message: 'Amount is not a number',
      });
    }

    if (total === Number(0)) {
      errors.push({
        type: 'amount',
        message: 'Amount cannot be zero',
      });
    }

    if (total > realBalance) {
      errors.push({
        type: 'balance',
        message: 'You do not have enough funds.',
      });
    }

    if (errors.length) {
      throw errors;
    }

    return true;
  };

  _getTransactionData = () => {
    const { coinid } = this.context;
    const { payments } = this.props;
    const { fee } = this.state;
    const isRBFEnabled = true;

    return new Promise((resolve, reject) => {
      try {
        if (this._verify()) {
          const transactionData = coinid.buildTransactionData(payments, fee, isRBFEnabled);
          const splitData = transactionData.split(':');
          this.savedUnsignedHex = splitData[3];
          return resolve(transactionData);
        }
      } catch (err) {
        Alert.alert(`${err}`);
        return reject(err);
      }
    });
  };

  _setFee = (fee) => {
    this.fee = fee;
    this._calculateTotal();
  };

  _onPressItem = (item) => {
    const { dialogNavigate } = this.context;
    const {
      navigation, balance, onAddToBatch, onRemoveFromBatch,
    } = this.props;

    dialogNavigate(
      'EditTransaction',
      {
        onAddToBatch,
        onRemoveFromBatch,
        balance,
        navigation,
        editItem: item,
        fee: this.fee,
      },
      this.context,
      false,
    );
  };

  _handleReturnData = (data) => {
    const { coinid } = this.context;
    const { payments, onQueuedTx } = this.props;
    const [action, hex] = data.split('/');

    const refresh = () => {
      // this.forceNoRender = true;
      onQueuedTx();
    };

    if (action === 'TX' && hex) {
      coinid
        .queueTx(hex, this.savedUnsignedHex)
        .then(queueData => coinid.noteHelper.saveNotes(queueData.tx, payments))
        .then(() => refresh())
        .catch((err) => {
          Alert.alert(`${err}`);
        });
    }
  };

  render() {
    const { subTotal, total, ticker } = this.state;
    const { payments } = this.props;

    const renderTransportContent = ({
      isSigning, signingText, cancel, submit,
    }) => {
      let validationError = [];
      try {
        this._verify();
      } catch (err) {
        validationError = err;
      }

      let disableButton = false;
      if (isSigning || validationError.length) {
        disableButton = true;
      }

      const renderError = () => {
        if (!validationError.length) {
          return null;
        }

        return validationError.map(({ message }, i) => (
          <Text key={i} style={styles.warningText}>
            {message}
          </Text>
        ));
      };

      return (
        <View style={styles.modalContent}>
          <View style={styles.batchedHeaderContainer}>
            <Text style={styles.batchedHeader}>Batched Transactions</Text>
          </View>
          <BatchList onPress={this._onPressItem} batchedTxs={payments} disabled={isSigning} />
          <View style={styles.summaryContainer}>
            <View style={{ marginBottom: 24, marginTop: 16 }}>
              <FeeSlider
                onChange={(val) => {
                  this._setFee(val);
                }}
                amount={subTotal}
                batchedTransactions={payments}
                disabled={isSigning}
              />
            </View>

            <RowInfo
              style={[{ marginBottom: 24 }]}
              childStyle={validationError.length ? styles.totalError : {}}
              title="Total"
            >
              {`${numFormat(total, ticker)} ${ticker}`}
            </RowInfo>
            {renderError()}
            <Button
              style={styles.formButton}
              onPress={submit}
              disabled={disableButton}
              isLoading={isSigning}
              loadingText={signingText}
            >
              Sign with COINiD
            </Button>
            <CancelButton show={isSigning} onPress={cancel} marginTop={16}>
              Cancel
            </CancelButton>
          </View>
        </View>
      );
    };

    return (
      <COINiDTransport
        getData={this._getTransactionData}
        handleReturnData={this._handleReturnData}
        parentDialog="Sign"
      >
        {renderTransportContent}
      </COINiDTransport>
    );
  }
}

export default Sign;
