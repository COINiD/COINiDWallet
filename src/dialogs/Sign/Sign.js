

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Alert, View } from 'react-native';
import Big from 'big.js';

import {
  Text, BatchList, RowInfo, FeeSlider, DetailsModal, COINiDTransport, Button, CancelButton,
} from '../../components';

import styles from './styles';
import { numFormat } from '../../utils/numFormat';

export default class Sign extends Component {
  constructor(props, context) {
    super(props);

    this.fee = 0;
    const { ticker } = context.coinid;

    this.state = {
      total: 0,
      subTotal: 0,
      fee: this.fee,
      ticker,
    };
  }

  getChildContext() {
    return {
      theme: this.props.theme ? this.props.theme : this.context.theme,
    };
  }

  componentDidMount() {
    this._calculateTotal();
  }

  componentWillReceiveProps() {
    this._calculateTotal();
  }

  shouldComponentUpdate() {
    if (this.forceNoRender) {
      return false;
    }
    return true;
  }

  _calculateTotal = () => {
    const { payments } = this.props;
    const { fee } = this;

    let subTotal = payments.reduce((a, c) => a.plus(c.amount), Big(0));
    const total = Number(subTotal.plus(fee));
    subTotal = Number(subTotal);

    this.setState({
      total,
      subTotal,
      fee,
    });
  }

  _verify = () => {
    let { total } = this.state;
    total = Number(total);

    const { payments, balance } = this.props;
    const errors = [];

    if (!payments.length)Â {
      errors.push({
        type: 'payments',
        message: 'no payments...',
      });
    }

    if (isNaN(total)) {
      errors.push({
        type: 'amount',
        message: 'amount is not a number',
      });
    }

    if (total === Number(0)) {
      errors.push({
        type: 'amount',
        message: 'amount cannot be zero',
      });
    }

    if (total > balance) {
      errors.push({
        type: 'balance',
        message: 'not enough funds',
      });
    }

    if (errors.length) {
      throw (errors);
    }

    return true;
  }

  _getTransactionData = () => {
    const { coinid } = this.context;
    const { payments } = this.props;
    const { fee } = this.state;
    const isRBFEnabled = true;

    return new Promise((resolve, reject) => {
      try {
        if (this._verify()) {
          const transactionData = coinid.buildTransactionData(payments, fee, isRBFEnabled);
          const splitData = transactionData.split(':');
          this.savedUnsignedHex = splitData[3];
          return resolve(transactionData);
        }
      } catch (err) {
        Alert.alert(`${err}`);
        return reject(err);
      }
    });
  }

  _setFee = (fee) => {
    this.fee = fee;
    this._calculateTotal();
  }

  _open = () => {
    this.forceNoRender = false;
    this.forceUpdate();
    this.elModal._open();
  }

  _close = () => {
    this.forceNoRender = true;
    this.elModal._close();
  }

  _onPressItem = (item) => {
    const { sendModal } = this.props;
    this._close();
    sendModal._open(item);
  }

  _onOpened = () => {
    const { onOpened } = this.props;
    onOpened();
  }

  _onClosed = () => {
    const { onClosed } = this.props;
    onClosed();
  }

  _handleReturnData = (data) => {
    const { coinid } = this.context;
    const { payments, onQueuedTx } = this.props;
    const [action, hex] = data.split('/');

    const refresh = () => {
      this.forceNoRender = true;
      onQueuedTx();
    };

    if (action === 'TX' && hex) {
      coinid
        .queueTx(hex, this.savedUnsignedHex)
        .then(queueData => coinid.noteHelper.saveNotes(queueData.tx, payments))
        .then(() => refresh())
        .catch((err) => {
          Alert.alert(`${err}`);
        });
    }
  };

  render() {
    const { subTotal, total, ticker } = this.state;
    const { payments } = this.props;

    const renderTransportContent = ({
      isSigning, signingText, cancel, submit,
    }) => {
      let validationError = [];
      try {
        this._verify();
      } catch (err) {
        validationError = err;
      }

      let disableButton = false;
      if (isSigning || validationError.length) {
        disableButton = true;
      }

      return (
        <View style={styles.modalContent}>
          <View style={styles.batchedHeaderContainer}>
            <Text style={styles.batchedHeader}>Batched Transactions</Text>
          </View>
          <BatchList
            onPress={this._onPressItem}
            batchedTxs={payments}
            disabled={isSigning}
          />
          <View style={styles.summaryContainer}>
            <View style={{ marginBottom: 24, marginTop: 16 }}>
              <FeeSlider
                onChange={(val) => { this._setFee(val); }}
                amount={subTotal}
                batchedTransactions={payments}
                disabled={isSigning}
              />
            </View>

            <RowInfo
              style={[{ marginBottom: 24 }]}
              childStyle={validationError.length ? { color: '#FA503C' } : {}}
              title="Total"
            >
              { `${numFormat(total, ticker)} ${ticker}` }
            </RowInfo>
            <Button
              style={styles.formButton}
              onPress={submit}
              disabled={disableButton}
              isLoading={isSigning}
              loadingText={signingText}
            >
              Sign with COINiD
            </Button>
            <CancelButton show={isSigning} onPress={cancel} marginTop={16}>
              Cancel
            </CancelButton>
          </View>
        </View>);
    };

    return (
      <DetailsModal
        ref={(c) => { this.elModal = c; }}
        title="Sign Transactions"
        verticalPosition="flex-end"
        onOpened={this._onOpened}
        onClosed={this._onClosed}
      >
        <COINiDTransport
          getData={this._getTransactionData}
          handleReturnData={this._handleReturnData}
        >
          { renderTransportContent }
        </COINiDTransport>
      </DetailsModal>
    );
  }
}

Sign.contextTypes = {
  coinid: PropTypes.object,
  type: PropTypes.string,
  theme: PropTypes.string,
};

Sign.childContextTypes = {
  theme: PropTypes.string,
};

Sign.propTypes = {
  payments: PropTypes.array,
  balance: PropTypes.number,
  theme: PropTypes.string,
};

Sign.defaultProps = {
  payments: [],
  balance: 0,
  theme: 'light',
};
