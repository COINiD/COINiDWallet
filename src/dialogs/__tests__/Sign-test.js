import 'react-native';
import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import PropTypes from 'prop-types';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import Sign from '../Sign';

import COINiDPublic from '../../libs/coinid-public';
import settings from '../../config/settings';
import storageHelper from '../../utils/storageHelper';
import settingHelper from '../../utils/settingHelper';

const context = {
  coinid: COINiDPublic(
    settings.coin,
    storageHelper(`${settings.coin}-hot`),
    `${settings.coin}-hot`,
  ),
  settingHelper: settingHelper(settings.coin),
}

describe('Sign', () => {
  const defaultProps = {
    payments: [{
      address: "msvEw9nCEtDtayUe9Mthh657yaDaqNN5A7",
      amount: "100",
      note: "woot",
    }],
    balance: 1000,
  };

  const getWrapper = (props=defaultProps)Â => {
    const wrapper = shallow(
      <Sign {...props} />,
      { context },
    );

    const instance = wrapper.instance();
    instance.amountRef = { _updateAmount: jest.fn() };

    return { wrapper, instance };
  }

  it('gets transaction data', () => {
    const { wrapper, instance } = getWrapper();
    expect(wrapper.state()).toMatchSnapshot();

    expect.assertions(2);
    return instance._getTransactionData()
      .then(
        () => {},
        err => expect(err).toEqual('Not enough funds!')
      );
  });

});
