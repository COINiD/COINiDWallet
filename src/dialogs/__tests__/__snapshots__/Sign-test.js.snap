// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign gets transaction data 1`] = `
Object {
  "fee": 0,
  "subTotal": 100,
  "ticker": "tBTC",
  "total": 100,
}
`;
